staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured/unstructured.go

```diff
@@ -310,6 +338,10 @@ func (u *Unstructured) GetLabels() map[string]string {
 }
 
 func (u *Unstructured) SetLabels(labels map[string]string) {
+	if labels == nil || len(labels) == 0 {
```

Comment:
`labels == nil` is not needed here and below.

Reply:
On a second thought: wouldn't the `omitempty` flag drop the field anyway when `len(labels)==0` holds? So in the output it does not make a difference. But if we go back to `== nil` only comparison, we keep the `GetLabels(SetLabels(labels)) == labels` invariant.

Reply:
The omitempty tag is present on the `ObjectMeta` struct.

`Unstructured` is just a simple map so it won't respect omitempty tags automatically.

Reply:
What I wanted to day: if this does not fix anything observable from the outside, let's keep the simpler programming model by not doing magical `len == 0 => nil` transformations.

Reply:
What about this:
Unstructured with nil fields -> json -> Unstructured
map[string]interface{} from first and second unstructured will not be the same. I.e. round tripping via json does not work?

Reply:
Hm, no, this is only the case when round tripped via typed struct with `omitempty` tags on fields. I think the behavior should match `omitempty` tags on structs where we know them. Any "api expert" we can ask here? :)

Reply:
Round tripping uses semantic comparison. And under that both are equal.

Reply:
I think unstructured should produce same json as if a typed struct was marshaled. Here is an example https://play.golang.org/p/ubGUOfntYIM - note how all fields with `omitempty` were omitted. Both `nil` and empty.

It is inconsistent that e.g. labels, if set to an empty map, will produce an empty json object in unstructured when the field is omitted if encoded via json package.

Reply:
> if this does not fix anything observable from the outside

@ash2k thanks for the clarification. This means that my assumption above is not true, i.e. we get a `true` value in JSON because we don't have `omitempty` tags in Unstructured.